draft-ietf-httpbis-priority.md

```diff
@@ -108,101 +114,100 @@ The term control stream is used to describe the HTTP/2 stream with identifier
-implicit signal to use the default priority. Third, the PRIORITY frame, which is
-always explicit since it always contains an Exclusive flag, Stream Dependency
-field, and Weight field.
-
-The rich flexibility of tree building is rarely exercised. Experience has shown
-that clients tend to choose a single model optimized for a web use case and
-experiment within the model constraints, or do nothing at all. Furthermore, many
-clients build their prioritization tree in a unique way, which makes it
-difficult for servers to understand their intent and act or intervene
-accordingly.
-
-Many HTTP/2 server implementations do not include support for the priority
-scheme. Some instead favor custom server-driven schemes based on heuristics or
-other hints, such as resource content type or request generation
-order. For example, a server, with knowledge of the document structure, might
-want to prioritize the delivery of images that are critical to user experience
-above other images, but below the CSS files. Since client trees vary, it is
-impossible for the server to determine how such images should be prioritized
-against other responses.
-
-The HTTP/2 scheme allows intermediaries to coalesce multiple client trees into a
+multiplexing is the ability to prioritize the sending of information.
+Prioritization is a difficult problem, so it will always be suboptimal,
+particularly if one endpoint operates in ignorance of the needs of its peer.
+Priority signalling allows endpoints to communicate their own view of priority
+needs, which can be combined with other factors that are considered during the
+peer's prioritization decision making.
+
+RFC 7540 stream priority ({{Section 5.3 of ?RFC7540}}) is a complex system where
+clients signal stream dependencies and weights to describe an unbalanced tree.
+It suffered from poor deployment and interoperability and was deprecated in a
+revision of HTTP/2 ({{HTTP2}}). However, in order to maintain wire
+compatibility, HTTP/2 priority signals are still mandatory to handle (see
+{{Section 5.3.2 of HTTP2}}).
+
+Clients can build RFC 7540 trees with rich flexibility but experience has shown
+this is rarely exercised. Instead they tend to choose a single model optimized
+for a single use case and experiment within the model constraints, or do nothing
+at all. Furthermore, many clients build their prioritization tree in a unique
+way, which makes it difficult for servers to understand their intent and act or
+intervene accordingly.
+
+Many RFC 7540 server implementations do not act on HTTP/2 priority signals. Some
+instead favor custom server-driven schemes based on heuristics or other hints,
+such as resource content type or request generation order. For example, a
+server, with knowledge of the document structure, might want to prioritize the
+delivery of images that are critical to user experience above other images, but
+below the CSS files. Since client trees vary, it is impossible for the server to
```

Comment:
I would re-structure this to be an example of the last sentence of the previous paragraph:

- Client trees are heterogeneous
- Makes it difficult to insert/reprioritize
- Consequently, some servers ignore client signals in favor of other heuristics to make these use cases easier

Now, these two somewhat come across as separate concepts, while to me, they are inherently linked, one the consequence of the other. 

(on a related note: I'm not sure the newly proposed system in this document really fully solves this issue, but that's another discussion :))

Reply:
The substantive comment here relates to old text that has been there for a long time, it's not suitable for fixing here. Please submit a new issue and/or PR if you feel it important to address.